packages/salesforcedx-vscode-core/src/orgBrowser/metadataOutlineProvider.ts

```diff
@@ -24,33 +37,144 @@ export class MetadataOutlineProvider
     this.defaultOrg = defaultOrg;
   }
 
-  public async refresh(defaultOrg?: string): Promise<void> {
-    if (!isNullOrUndefined(defaultOrg)) {
-      this.defaultOrg = defaultOrg;
+  public async onViewChange() {
+    if (await this.getDefaultUsernameOrAlias()) {
+      this.internalOnDidChangeTreeData.fire();
     }
+  }
+
+  public async refresh(defaultOrg?: string): Promise<void> {
+    await this.getDefaultUsernameOrAlias();
     this.internalOnDidChangeTreeData.fire();
   }
 
   public getTreeItem(element: BrowserNode): vscode.TreeItem {
     return element;
   }
 
-  public getChildren(element?: BrowserNode): Promise<BrowserNode[]> {
+  public async getChildren(element?: BrowserNode): Promise<BrowserNode[]> {
     if (isNullOrUndefined(element)) {
       if (!isNullOrUndefined(this.defaultOrg)) {
         const org = new BrowserNode(this.defaultOrg, NodeType.Org);
         return Promise.resolve([org]);
       } else {
-        return Promise.resolve([]);
+        const emptyDefault = new BrowserNode(
+          'No default org set',
+          NodeType.EmptyNode
+        );
+        return Promise.resolve([emptyDefault]);
+      }
+    } else if (element.type === NodeType.Org) {
+      const metadataTypes = await this.getTypes();
+      element.children = metadataTypes;
+      return Promise.resolve(element.children);
+    } else if (element.type === NodeType.MetadataType) {
+      const metadataCmps = await this.getComponents(element);
+      if (isNullOrUndefined(metadataCmps) || metadataCmps.length === 0) {
+        const emptyDefault = new BrowserNode(
+          'No components available',
```

Comment:
Do we want to show a node if there aren't any components? In the normal file tree if there aren't any files in a folder nothing is shown.

Reply:
I'm leaning towards displaying a node with a `No components available` label because I found it difficult to easily see that the type node was expanded and no components were below it once the data had been requested.

cc @ntotten Thoughts?

Reply:
I don't have a strong opinion on this. The file tree in VS Code allows you to click expand on the node and just doesn't show anything when the folder is empty. In this case though since we are loading data from the org I almost wonder if not showing anything will make it feel like an err. I'm okay with the current design.

Reply:
I'll leave the message in for now and we can adjust if we get feedback on it. Thanks!